### ISSUE TITLE
GCC 7.3.0 issue

### ISSUE THREAD

**Issue by carefulguy** [nasa/trick#610]
Hey Trick guys

\When I trick-CP with GCC 7.3.0 (standard with new Ubuntu 18), I get a bunch of compiler errors like:

Example:
/usr/include/c++/7/bits/nested_exception.h:65:7: error: use of undeclared identifier 'noreturn'
    [[noreturn]]
...

After all the errors though it did succeed with "Trick Build Process Complete" so maybe it doesn't matter.

What gcc version is best, and how do I change which gcc after Trick is installed?

**Comment by iamthad** [comment]
Hmm, which version of LLVM do you have installed? I wonder if it is too old.

**Comment by carefulguy** [comment]
clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
Target: x86_64-pc-linux-gnu

**Comment by carefulguy** [comment]
No, I'm actually using the latest and greatest that comes with Ubuntu 18 (are you surprised Thad? ;)

I suspect Trick hasn't caught up to it yet, which other issues seem to indicate (#579).  Since It still compiles at the end - I'll close this as a duplicate.

**Comment by Sstewart26** [comment]
I got this same error compiling SIM_ball_L1 from Trick 17.3 using clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final).  It is strange because there are number build errors, but I get a S_main_Linux__x86_64.exe and it seems to run.  Appears to happen when running ICG.

Running    ICG
In file included from /Trick/trick_sims/Ball/SIM_ball_L1/S_source.hh:10:
In file included from /usr/include/c++/7/new:40:
In file included from /usr/include/c++/7/exception:143:
/usr/include/c++/7/bits/nested_exception.h:65:7: error: use of undeclared identifier 'noreturn'
    [[noreturn]]
      ^
/usr/include/c++/7/bits/nested_exception.h:65:16: error: expected body of lambda expression
    [[noreturn]]
               ^
/usr/include/c++/7/bits/nested_exception.h:66:5: error: expected member name or ';' after declaration specifiers
    void
    ^
/usr/include/c++/7/bits/nested_exception.h:110:7: error: use of undeclared identifier 'noreturn'
    [[noreturn]]
      ^
/usr/include/c++/7/bits/nested_exception.h:110:16: error: expected body of lambda expression
    [[noreturn]]
               ^
/usr/include/c++/7/bits/nested_exception.h:111:5: error: expected unqualified-id
    inline void
    ^
/usr/include/c++/7/bits/nested_exception.h:134:12: error: no template named '__rethrow_if_nested_cond'
    inline __rethrow_if_nested_cond<_Ex>
           ^
/usr/include/c++/7/bits/nested_exception.h:138:12: error: no member named 'rethrow_nested' in 'std::nested_exception'
        __ne_ptr->rethrow_nested();
        ~~~~~~~~  ^
In file included from /Trick/trick_sims/Ball/SIM_ball_L1/S_source.hh:17:
In file included from /Trick/include/trick/ClassSizeCheck.hh:9:
In file included from /usr/include/c++/7/map:60:
In file included from /usr/include/c++/7/bits/stl_tree.h:69:
/usr/include/c++/7/ext/aligned_buffer.h:54:7: error: expected member name or ';' after declaration specifiers
      alignas(__alignof__(_Tp2::_M_t)) unsigned char _M_storage[sizeof(_Tp)];
      ^
/usr/include/c++/7/ext/aligned_buffer.h:63:36: error: use of undeclared identifier '_M_storage'
      { return static_cast<void*>(&_M_storage); }
                                   ^
/usr/include/c++/7/ext/aligned_buffer.h:67:42: error: use of undeclared identifier '_M_storage'
      { return static_cast<const void*>(&_M_storage); }
                                         ^
In file included from /Trick/trick_sims/Ball/SIM_ball_L1/S_source.hh:35:
In file included from /Trick/include/trick/EventManager.hh:22:
In file included from /Trick/include/trick/Event.hh:7:
In file included from /Trick/include/trick/mm_macros.hh:8:
In file included from /Trick/include/trick/MemoryManager.hh:16:
In file included from /usr/include/c++/7/functional:58:
/usr/include/c++/7/bits/std_function.h:104:3: error: declaration of anonymous union must be a definition
  union [[gnu::may_alias]] _Any_data
  ^
/usr/include/c++/7/bits/std_function.h:170:23: error: unknown type name '_Any_data'
        _M_get_pointer(const _Any_data& __source)
                             ^
/usr/include/c++/7/bits/std_function.h:181:11: error: unknown type name '_Any_data'
        _M_clone(_Any_data& __dest, const _Any_data& __source, true_type)
                 ^
/usr/include/c++/7/bits/std_function.h:181:36: error: unknown type name '_Any_data'
        _M_clone(_Any_data& __dest, const _Any_data& __source, true_type)
                                          ^
/usr/include/c++/7/bits/std_function.h:189:11: error: unknown type name '_Any_data'
        _M_clone(_Any_data& __dest, const _Any_data& __source, false_type)
                 ^
/usr/include/c++/7/bits/std_function.h:189:36: error: unknown type name '_Any_data'
        _M_clone(_Any_data& __dest, const _Any_data& __source, false_type)
                                          ^
/usr/include/c++/7/bits/std_function.h:198:13: error: unknown type name '_Any_data'
        _M_destroy(_Any_data& __victim, true_type)
                   ^
/usr/include/c++/7/bits/std_function.h:205:13: error: unknown type name '_Any_data'
        _M_destroy(_Any_data& __victim, false_type)
                   ^
/usr/include/c++/7/bits/std_function.h:212:13: error: unknown type name '_Any_data'
        _M_manager(_Any_data& __dest, const _Any_data& __source,
                   ^
/usr/include/c++/7/bits/std_function.h:212:38: error: unknown type name '_Any_data'
        _M_manager(_Any_data& __dest, const _Any_data& __source,
                                            ^
/usr/include/c++/7/bits/std_function.h:238:18: error: unknown type name '_Any_data'
        _M_init_functor(_Any_data& __functor, _Functor&& __f)
                        ^
/usr/include/c++/7/bits/std_function.h:263:18: error: unknown type name '_Any_data'
        _M_init_functor(_Any_data& __functor, _Functor&& __f, true_type)
                        ^
/usr/include/c++/7/bits/std_function.h:267:18: error: unknown type name '_Any_data'
        _M_init_functor(_Any_data& __functor, _Functor&& __f, false_type)
                        ^
/usr/include/c++/7/bits/std_function.h:281:35: error: unknown type name '_Any_data'
    typedef bool (*_Manager_type)(_Any_data&, const _Any_data&,
                                  ^
/usr/include/c++/7/bits/std_function.h:281:53: error: unknown type name '_Any_data'
    typedef bool (*_Manager_type)(_Any_data&, const _Any_data&,
                                                    ^
/usr/include/c++/7/bits/std_function.h:284:5: error: unknown type name '_Any_data'
    _Any_data     _M_functor;
    ^
/usr/include/c++/7/bits/std_function.h:173:60: error: '_Functor' does not refer to a value
            __stored_locally? std::__addressof(__source._M_access<_Functor>())
                                                                  ^
/usr/include/c++/7/bits/std_function.h:156:23: note: declared here
    template<typename _Functor>
                      ^
/usr/include/c++/7/bits/std_function.h:173:70: error: expected expression
            __stored_locally? std::__addressof(__source._M_access<_Functor>())
                                                                            ^
/usr/include/c++/7/bits/std_function.h:174:55: error: '_Functor' does not refer to a value
            /* have stored a pointer */ : __source._M_access<_Functor*>();
                                                             ^
/usr/include/c++/7/bits/std_function.h:156:23: note: declared here
    template<typename _Functor>
                      ^
/usr/include/c++/7/bits/std_function.h:174:64: error: expected expression
            /* have stored a pointer */ : __source._M_access<_Functor*>();
                                                                      ^
/usr/include/c++/7/bits/std_function.h:174:66: error: expected expression
            /* have stored a pointer */ : __source._M_access<_Functor*>();
                                                                        ^
/usr/include/c++/7/bits/std_function.h:191:21: error: '_Functor' does not refer to a value
          __dest._M_access<_Functor*>() =
                           ^
/usr/include/c++/7/bits/std_function.h:156:23: note: declared here
    template<typename _Functor>
                      ^
/usr/include/c++/7/bits/std_function.h:191:30: error: expected expression
          __dest._M_access<_Functor*>() =
                                    ^
/usr/include/c++/7/bits/std_function.h:191:32: error: expected expression
          __dest._M_access<_Functor*>() =
                                      ^
/usr/include/c++/7/bits/std_function.h:192:39: error: '_Functor' does not refer to a value
            new _Functor(*__source._M_access<_Functor*>());
                                             ^
/usr/include/c++/7/bits/std_function.h:156:23: note: declared here
    template<typename _Functor>
                      ^
/usr/include/c++/7/bits/std_function.h:192:48: error: expected expression
            new _Functor(*__source._M_access<_Functor*>());
                                                      ^
/usr/include/c++/7/bits/std_function.h:192:50: error: expected expression
            new _Functor(*__source._M_access<_Functor*>());
                                                        ^
/usr/include/c++/7/bits/std_function.h:200:23: error: '_Functor' does not refer to a value
          __victim._M_access<_Functor>().~_Functor();
                             ^
/usr/include/c++/7/bits/std_function.h:156:23: note: declared here
    template<typename _Functor>
                      ^
/usr/include/c++/7/bits/std_function.h:200:33: error: expected expression
          __victim._M_access<_Functor>().~_Functor();
                                       ^
/usr/include/c++/7/bits/std_function.h:207:30: error: '_Functor' does not refer to a value
          delete __victim._M_access<_Functor*>();
                                    ^
/usr/include/c++/7/bits/std_function.h:156:23: note: declared here
    template<typename _Functor>
                      ^
/usr/include/c++/7/bits/std_function.h:207:39: error: expected expression
          delete __victim._M_access<_Functor*>();
                                             ^
/usr/include/c++/7/bits/std_function.h:207:41: error: expected expression
          delete __victim._M_access<_Functor*>();
                                               ^
/usr/include/c++/7/bits/std_function.h:219:25: error: expected expression
              __dest._M_access<const type_info*>() = &typeid(_Functor);
                               ^
/usr/include/c++/7/bits/std_function.h:223:25: error: '_Functor' does not refer to a value
              __dest._M_access<_Functor*>() = _M_get_pointer(__source);
                               ^
/usr/include/c++/7/bits/std_function.h:156:23: note: declared here
    template<typename _Functor>
                      ^
/usr/include/c++/7/bits/std_function.h:223:34: error: expected expression
              __dest._M_access<_Functor*>() = _M_get_pointer(__source);
                                        ^
/usr/include/c++/7/bits/std_function.h:223:36: error: expected expression
              __dest._M_access<_Functor*>() = _M_get_pointer(__source);
                                          ^
/usr/include/c++/7/bits/std_function.h:268:24: error: '_Functor' does not refer to a value
        { __functor._M_access<_Functor*>() = new _Functor(std::move(__f)); }
                              ^
/usr/include/c++/7/bits/std_function.h:156:23: note: declared here
    template<typename _Functor>
                      ^
/usr/include/c++/7/bits/std_function.h:268:33: error: expected expression
        { __functor._M_access<_Functor*>() = new _Functor(std::move(__f)); }
                                       ^
/usr/include/c++/7/bits/std_function.h:268:35: error: expected expression
        { __functor._M_access<_Functor*>() = new _Functor(std::move(__f)); }
                                         ^
/usr/include/c++/7/bits/std_function.h:299:23: error: unknown type name '_Any_data'
      _M_invoke(const _Any_data& __functor, _ArgTypes&&... __args)
                      ^
/usr/include/c++/7/bits/std_function.h:314:23: error: unknown type name '_Any_data'
      _M_invoke(const _Any_data& __functor, _ArgTypes&&... __args)
                      ^
/usr/include/c++/7/bits/std_function.h:331:23: error: unknown type name '_Any_data'
      _M_invoke(const _Any_data& __functor, _ArgTypes&&... __args)
                      ^
/usr/include/c++/7/bits/std_function.h:349:18: error: unknown type name '_Any_data'
      _M_manager(_Any_data& __dest, const _Any_data& __source,
                 ^
/usr/include/c++/7/bits/std_function.h:349:43: error: unknown type name '_Any_data'
      _M_manager(_Any_data& __dest, const _Any_data& __source,
                                          ^
/usr/include/c++/7/bits/std_function.h:371:23: error: unknown type name '_Any_data'
      _M_invoke(const _Any_data& __functor, _ArgTypes&&... __args)
                      ^
/usr/include/c++/7/bits/std_function.h:356:23: error: expected expression
            __dest._M_access<const type_info*>() = &typeid(_Functor);
                             ^
/usr/include/c++/7/bits/std_function.h:360:23: error: unexpected type name '_Functor': expected expression
            __dest._M_access<_Functor*>() =
                             ^
/usr/include/c++/7/bits/std_function.h:360:32: error: expected expression
            __dest._M_access<_Functor*>() =
                                      ^
/usr/include/c++/7/bits/std_function.h:360:34: error: expected expression
            __dest._M_access<_Functor*>() =
                                        ^
/usr/include/c++/7/bits/std_function.h:628:44: error: unknown type name '_Any_data'
      using _Invoker_type = _Res (*)(const _Any_data&, _ArgTypes&&...);
                                           ^
/usr/include/c++/7/bits/std_function.h:717:4: error: unknown type name '_Any_data'
          _Any_data __typeinfo_result;
          ^
/usr/include/c++/7/bits/std_function.h:719:40: error: expected expression
          return *__typeinfo_result._M_access<const type_info*>();
                                              ^
/usr/include/c++/7/bits/std_function.h:744:6: error: unknown type name '_Any_data'
            _Any_data __ptr;
            ^
/usr/include/c++/7/bits/std_function.h:746:29: error: expected expression
            return __ptr._M_access<const _Functor*>();
                                   ^
Writing    build/Trick/trick_sims/Ball/models/ball/L1/include/io_ball_force.cpp
Writing    build/Trick/trick_sims/Ball/models/ball/L1/include/io_ball_state.cpp
Writing    build/Trick/trick_sims/Ball/SIM_ball_L1/io_S_source.cpp
Writing    Makefile_io_src
Writing    Makefile_src
NewDep     /Trick/trick_sims/Ball/SIM_ball_L1/S_source.hh
DepTracing /Trick/trick_sims/Ball/SIM_ball_L1/S_source.hh
DepTracing /Trick/trick_sims/Ball/models/ball/L1/include/ball_force.h
DepTracing /Trick/trick_sims/Ball/models/ball/L1/include/ball_state.h
DepTracing /Trick/trick_sims/Ball/models/ball/L1/include/ball_proto.h
DepTracing  /Trick/trick_sims/Ball/models/ball/L1/src/ball_force_default_data.c
DepTracing  /Trick/trick_sims/Ball/models/ball/L1/src/ball_force_field.c
DepTracing  /Trick/trick_sims/Ball/models/ball/L1/src/ball_print.c
DepTracing  /Trick/trick_sims/Ball/models/ball/L1/src/ball_state_deriv.c
DepTracing  /Trick/trick_sims/Ball/models/ball/L1/src/ball_state_init.c
DepTracing  /Trick/trick_sims/Ball/models/ball/L1/src/ball_state_integ.c
DepTracing  /Trick/trick_sims/Ball/models/ball/L1/src/ball_state_alt_integ.c
DepTracing  /Trick/trick_sims/Ball/models/ball/L1/src/ball_state_default_data.c
Compiling  build/S_source.cpp
Compiling  /Trick/trick_sims/Ball/models/ball/L1/src/ball_force_default_data.c
Compiling  /Trick/trick_sims/Ball/models/ball/L1/src/ball_force_field.c
Compiling  /Trick/trick_sims/Ball/models/ball/L1/src/ball_print.c
Compiling  /Trick/trick_sims/Ball/models/ball/L1/src/ball_state_alt_integ.c
Compiling  /Trick/trick_sims/Ball/models/ball/L1/src/ball_state_default_data.c
Compiling  /Trick/trick_sims/Ball/models/ball/L1/src/ball_state_deriv.c
Compiling  /Trick/trick_sims/Ball/models/ball/L1/src/ball_state_init.c
Compiling  /Trick/trick_sims/Ball/models/ball/L1/src/ball_state_integ.c
Compiling  build/Trick/trick_sims/Ball/SIM_ball_L1/io_S_source.cpp
Compiling  build/Trick/trick_sims/Ball/models/ball/L1/include/io_ball_force.cpp
Compiling  build/Trick/trick_sims/Ball/models/ball/L1/include/io_ball_state.cpp
Compiling  build/class_map.cpp
Writing    build/Trick/trick_sims/Ball/SIM_ball_L1/S_source_py.i
Writing    build/Trick/trick_sims/Ball/models/ball/L1/include/ball_force_py.i
Writing    build/Trick/trick_sims/Ball/models/ball/L1/include/ball_proto_py.i
Writing    build/Trick/trick_sims/Ball/models/ball/L1/include/ball_state_py.i
SWIGing    build/Trick/trick_sims/Ball/SIM_ball_L1/S_source_py.i
Compiling  build/Trick/trick_sims/Ball/SIM_ball_L1/S_source_py.cpp
SWIGing    build/Trick/trick_sims/Ball/models/ball/L1/include/ball_force_py.i
Compiling  build/Trick/trick_sims/Ball/models/ball/L1/include/ball_force_py.cpp
SWIGing    build/Trick/trick_sims/Ball/models/ball/L1/include/ball_proto_py.i
Compiling  build/Trick/trick_sims/Ball/models/ball/L1/include/ball_proto_py.cpp
SWIGing    build/Trick/trick_sims/Ball/models/ball/L1/include/ball_state_py.i
Compiling  build/Trick/trick_sims/Ball/models/ball/L1/include/ball_state_py.cpp
SWIGing    build/top.i
Compiling  build/top.cpp
Compiling  build/init_swig_modules.cpp
Linking    S_main_Linux__x86_64.exe
Writing    S_sie.resource
./S_main_Linux__x86_64.exe sie
Trick Build Process Complete

**Comment by alexlin0** [comment]
@Sstewart26, This was fixed after 17.3 came out.  If you don't want to see the errors, I think you'll have to move to the development version or wait until our next release, which should be this month.
